pages/create.unit.page.ts

```diff
@@ -0,0 +1,554 @@
+            }
+            else if(tabNames[i] === 'Послуги') {
+                const customServicesTitle  = this.page.locator('div[class*="ServicesUnitFlow_title"]');
+                const customServicesSearchInput = this.page.locator('div[class*="ServicesUnitFlow_searchInput"]');
+
+                await expect(customServicesTitle).toBeVisible();
+                await expect(customServicesSearchInput).toBeVisible();
+            }
+            else if(tabNames[i] === 'Вартість') {
+                const customPricesTitle  = this.page.locator('div[class*="PricesUnitFlow_title"]');
+                const paymentTypeSelect = this.page.locator('div[data-testid="div_CustomSelect"]');
+                const customUnitPrice = this.page.locator('div[data-testid="input_wrapper_RowUnitPrice"]').first();
+
+                await expect(customPricesTitle).toBeVisible();
+                await expect(paymentTypeSelect).toBeVisible();
+                await expect(customUnitPrice).toBeVisible();
+            }
+            else if(tabNames[i] === 'Контакти') {
+                const contactsTitle  = this.page.locator('div[class*="AuthContactCard_title"]');
+                const contactsCard = this.page.locator('div[class*="AuthContactCard_infoWrapper"]');
+
+                await expect(contactsTitle).toBeVisible();
+                await expect(contactsCard).toBeVisible();
+            }
+        }
+    }
+
+    async checkSection(sectionTitleLocator: Locator, sectionInputLocator: Locator, expectedTitleText: string, expectedInputBgText?: string, hasAsterisk?: boolean) {
+        await this.page.waitForLoadState('domcontentloaded');
+        await expect(sectionInputLocator).toBeVisible();
+    
+        const tagName = await sectionInputLocator.evaluate((element) => element.tagName.toLowerCase());
+
+        let sectionBgText;
+    
+        if(tagName === 'div') {
+            sectionBgText = await sectionInputLocator.innerText();
+            await expect(this.categoriesDropDownArrowDown).toBeVisible();
+        } else if(tagName === 'input') {
+            sectionBgText = await sectionInputLocator.getAttribute('placeholder'); 
+        } else if(sectionInputLocator === this.vehicleManufacturerInput) {
+            await expect(this.vehicleManufacturerInputSearchIcon).toBeVisible()
+        }
+
+        const sectionTitleText = await sectionTitleLocator.innerText();
+
+        if (hasAsterisk) {
+            await expect(sectionTitleText).toContain(expectedTitleText);
+            await expect(sectionTitleText).toContain('*');
+        } else {
+            await expect(sectionTitleText).toContain(expectedTitleText);
+        }
+        if (sectionBgText !== undefined && sectionBgText !== null) {
+            await expect(sectionBgText).toContain(expectedInputBgText);
+        } 
+    }
+
+    async clickOnNextBtn() {
+        await this.nextBtn.click();
+        await this.page.waitForLoadState('networkidle');
```

Comment:
Using waitForLoadState("networkidle") is a good practice, but consider using specific waits for elements to appear or be hidden when possible to make tests more robust against network delays.

Reply:
In this case, there is no specific element that I could use for waiting. So, I've decided to use waitForLoadState("networkidle")